go/fixchain/hash.go

```diff
@@ -0,0 +1,60 @@
+package fixchain
+
+import (
+	"crypto/sha256"
+	"sort"
+
+	"github.com/google/certificate-transparency/go/x509"
+)
+
+const hashSize = sha256.Size
+
+var newHash = sha256.New
+
+func hash(c *x509.Certificate) (hash [hashSize]byte) {
+	copy(hash[:], newHash().Sum(c.Raw))
+	return
+}
+
+func hashChain(ch []*x509.Certificate) (hash [hashSize]byte) {
+	h := newHash()
+	for _, c := range ch {
+		h.Write(newHash().Sum(c.Raw))
+	}
+	copy(hash[:], h.Sum(nil))
+	return
+}
+
+func hashBag(chain []*x509.Certificate) [hashSize]byte {
```

Comment:
Its a precondition that the chain is de-duped, I assume? I forget why we need a bag hash...

Anyway, assuming that it is a precondition, should we verify that? If it isn't, I am confused. :-)